### DISCUSSION TITLE
Help, my github actions not working<title>

### DISCUSSION THREAD

**Discussion by Retji** [community/community#38842] [Actions]
### Select Topic Area

Bug

### Body

Please I set up a github action to run a python file whose code is to scrape data from twitter and save as csv.

Somehow my yaml code has some error and I seem not to get it right.

Please if anyone can look at my code and point me to where I got it wrong I will appreciate.

name:   run campaign_tweets.py

# a little info on cron
# ┌────────── minute (0 - 59)
# │ ┌────────── hour (0 - 23)
# │ │ ┌────────── day of the month (1 - 31)
# │ │ │ ┌────────── month (1 - 12)
# │ │ │ │ ┌────────── day of the week (0 - 6)
# │ │ │ │ │                                  
# │ │ │ │ │
# │ │ │ │ │
# * * * * *

on:
  schedule:
    cron: '0*/12***' #to run the action every 12 hours
jobs:
  build:
    runs-on: ubuntu-latest

    # Steps represent a sequence of tasks that will be executed as part of the workflow
    steps:
          - name: checkout repo content
            uses: actions/checkout@v2 # checkout the repository content to github runner.

          - name: setup python
            uses: actions/setup-python@v4
            with:
              python-version: '3.9' # to install python version 3.9

          - name: setup dependencies and packages
            run: |
              python -m pip install --upgrade pip
              pip install snscrape
              pip install pandas
              pip install path
              pip instal csv
              pip install -r requirements.txt
          - name: execute py script # run the scrape_campaign_tweets.py to get the latest data
            env: 
                CONSUMER_KEY: ${{ secrets.CONSUMER_KEY }}
                CONSUMER_SECRET: ${{ secrets.CONSUMER_SECRET }}

            run: |
              campaign_tweets.py
          - name: Commit and push if it changed
            run: |
              git config user.name "${GITHUB_ACTOR}"
              git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
              git add -A
              timestamp=$(date -u)
              git commit -m "Scrapped: ${timestamp} (auto)"  
              git push "https://${GITHUB_ACTOR}:${TOKEN}@github.com/${GITHUB_REPOSITORY}.git" HEAD:main || exit 0

Details on the repository is on
 https://github.com/Retji/campaign_tweets/

**Comment by airtower-luna** [comment]
I had to look at the repository to see the formatting of the workflow, indentation is critical in YAML, so please remember to use a Markdown code block for posting code in the future!

Anyway, there are two problems in the trigger definition:

```yaml
on:
  schedule:
    cron: '0*/12***' #to run the action every 12 hours
```

1. `schedule` must be a list, and each element of the list must be a map that contains a `cron` key. Each element in a list is marked with a dash (`-`), just like in `steps`.
2. The value for your `cron` lacks spaces between the fields for minute, hour, and so on.

This should work:

```yaml
on:
  schedule:
    - cron: '0 */12 * * *' # to run the action every 12 hours
```

However I'd recommend picking a random minute instead of 0. Experience shows a lot of people pick the very start of the hour, and your job might be delayed (or even missed) during load peaks. :wink:
